### PR TITLE
Fixes stun times.

### PR DESCRIPTION
**Pull request by AnturK**
Or does it.

### CODE REVIEW

File: code/modules/mob/status_procs.dm [hunk 1]
```diff
@@ -27,7 +27,7 @@
 			S.duration = world.time + max(amount, remaining_duration)
 		else if(amount > 0)
 			S = apply_status_effect(STATUS_EFFECT_STUN)
-			S.duration = amount
+			S.duration = world.time + amount
```
Comment (ChangelingRain) [inline code/modules/mob/status_procs.dm]: This will not fix the problem, as it is set to world.time + duration in on_apply on the tick afterwards.

**Comment by ChangelingRain** [comment]
This doesn't fix it, #28696 does, etc

**Comment by AnturK** [comment]
That's even worse, why would you have invalid state for a prolonged amount of time anywhere in normal operation.

**Comment by ChangelingRain** [comment]
Yes, I realized that, that's why #28696 [adds all these checks for if it's processing to make absolutely sure that it's being given appropriate values for its current state.](https://github.com/tgstation/tgstation/pull/28696/files#diff-68d3c6f62aae0810e3865134ba55b8b5R26)
Which was, for reference, the source of the bug.

**Pull request closed without merging**